connectors/http/http-base/src/main/java/io/camunda/connector/http/base/utils/JsonHelper.java

```diff
@@ -19,26 +19,33 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.JsonParseException;
+import io.camunda.connector.api.json.ConnectorsObjectMapperSupplier;
+import java.io.IOException;
 import java.util.Optional;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class JsonHelper {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(JsonHelper.class);
+  public static JsonNode getAsJsonElement(Object body, final ObjectMapper mapper)
+      throws JsonProcessingException {
+    if (body instanceof String stringBody) {
+      return isJsonStringValid(stringBody) ? mapper.readTree(stringBody) : null;
+    } else {
+      return Optional.ofNullable(body).map(mapper::<JsonNode>valueToTree).orElse(null);
+    }
+  }
+
+  public static boolean isJsonStringValid(String jsonString) {
+    try {
+      ObjectMapper objectMapper = ConnectorsObjectMapperSupplier.DEFAULT_MAPPER;
+      JsonNode jsonNode = objectMapper.readTree(jsonString);
+      return jsonNode.isObject() || jsonNode.isArray();
+    } catch (JsonParseException | IOException e) {
+      return false;
+    }
+  }
 
-  public static JsonNode getAsJsonElement(final String strResponse, final ObjectMapper mapper) {
-    return Optional.ofNullable(strResponse)
-        .filter(response -> !response.isBlank())
-        .map(
-            response -> {
-              try {
-                return mapper.readTree(response);
-              } catch (JsonProcessingException e) {
-                LOGGER.error("Wasn't able to create a JSON node from string: " + strResponse);
-                throw new RuntimeException(e);
-              }
-            })
-        .orElse(null);
+  public static boolean isJsonValid(Object maybeJson) throws JsonProcessingException {
+    return getAsJsonElement(maybeJson, ConnectorsObjectMapperSupplier.DEFAULT_MAPPER) != null;
```

Comment:
Two of the three static methods directly access the object mapper via the static `ConnectorsObjectMapperSupplier.DEFAULT_MAPPER` call while the other has a parameter for it. Can we maybe refactors this utility to be passed into the class via parameter to avoid the static object mapper calls?

Reply:
Fixed ✅